packages/bierzo-wallet/src/routes/transactions/components/sorting.ts

```diff
@@ -0,0 +1,45 @@
+import { ProcessedTx } from '../../../store/notifications';
+
+export interface SortingStateProps {
+  readonly onSort: (orderBy: TxsOrder, order: SortOrder) => () => void;
+  readonly orderBy: TxsOrder;
+  readonly order: SortOrder;
+}
+
+export const ORDER_ASC = 1;
+export const ORDER_DESC = -1;
+export type SortOrder = 1 | -1;
+
+export const TX_AMOUNT_COLUMN = 'Amount';
+export const TX_DATE_COLUMN = 'Date';
+export type TxsOrder = 'Date';
+
+export const filterTxsBy = (
+  txs: ReadonlyArray<ProcessedTx>,
+  rowsPerPage: number,
+  pageNumber: number,
+  orderBy: TxsOrder,
+  order: SortOrder,
+): ReadonlyArray<ProcessedTx> => {
+  const orderedTxs = txs.slice(0).sort((a: ProcessedTx | undefined, b: ProcessedTx | undefined) => {
+    if (!a || !b) {
+      return 0;
+    }
+
+    if (orderBy === TX_DATE_COLUMN) {
+      return (a.time < b.time ? -1 : a.time > b.time ? 1 : 0) * order;
+    }
+
+    return 0;
```

Comment:
By default elements are equal? This does not make sense. Better avoid calling `.sort()` when you don't want to sort

Reply:
The `sort` method expects a function which returns a number, I have checked and this return never will be reached  but left the default just in case to tell to the algorithm: "Ey, just leave the order of those two elements in the same position as they are"

Is there any problem with this approach which I am missing?

Reply:
Yes, there is a difference between
1. Telling the sort algorithm all elements are equal (by returning 0)
2. All elements should be returned in the same order
Most sort algorithms will return a list that is entirely mixed when all elements are equal. This is also the JS standard. V8 implements a sort algorithm that is _stable_, i.e. if two elements are equal they remain in the original order. But you cannot assume you have this property.

Reply:
> by " all elements"

you mean the two elements which are being compared, right?

🤔 so, what should I do? throw an exception? or just leave as it is for safety? Any other idea?

Reply:
> you mean the two elements which are being compared, right?

Yes, if the two compared elements stay in the original order, the whole list styley in the original order.

I suggest not running the sort when you want to keep the original order:

```ts
  const orderedTxs = txs.slice(0); // make a mutable copy
  
  if (orderBy === TX_DATE_COLUMN) {
    orderedTxs.sort((a: ProcessedTx, b: ProcessedTx) => {
      return (a.time < b.time ? -1 : a.time > b.time ? 1 : 0) * order;
    }
  }
```

Reply:
Done!